lib/src/config/config.rs

```diff
@@ -847,7 +859,24 @@ impl ClientConfigBuilder {
         // Configure validator
         #[cfg(feature="validator")] {
             if config_file.validator.is_some() {
-                self.validator = Some(Some(ValidatorConfig {}));
+                let wallet_private_key = config_file.validator.as_ref()
+                    .map(|settings| settings.clone().wallet_private_key)
+                    .expect("Failed to load validator settings");
+
+                let wallet_key: Option<keys::KeyPair>;
+                if wallet_private_key.is_some() {
+                    let config_private_key = wallet_private_key.unwrap();
+                    wallet_key = Some(keys::KeyPair::from(
+                        keys::PrivateKey::from_str(&config_private_key)
+                            .map_err(|e| Error::config_error(format!("Invalid wallet private key: {:?}: {}", &config_private_key, e)))?
+                    ));
+                } else {
+                    wallet_key = None;
+                }
```

Comment:
Here's the more Rusty way to do this.
```suggestion
                let wallet_key = wallet_private_key.map(|config_private_key| {
                    let config_private_key = wallet_private_key.unwrap();
                    Some(keys::KeyPair::from(
                        keys::PrivateKey::from_str(&config_private_key)
                            .map_err(|e| Error::config_error(format!("Invalid wallet private key: {:?}: {}", &config_private_key, e)))?
                    ))
                });
```